### ISSUE TITLE
DNS issues on macOS 10.12.2 [Update: but caused by pfSense 2.3.2 configuration changes]

### ISSUE THREAD

**Issue by AniX** [Tunnelblick/Tunnelblick#351]
Last time I know that my configuration worked was under 10.11.5 with Tunnelblick 3.8 a few weeks ago. Not sure if I had to use Tunnelblick after upgrading to 10.12.

However, now after upgrading to 10.12.2, I still can connect to the VPN server, but my DNS seems to be broken as soon as I connect to the VPN.

I can ping public IP addresses and IP addresses in the remote VPN network, I'm connected to. But I cannot resolve any host names (e.g. nslookup, normal browser address), no matter if public hosts or hosts in the remote VPN network.

Soon after connecting, Tunnelblick shows a warning:
"After connecting to office new, DNS does not appear to be working. This may mean that your VPN is not configured correctly."

Under macOS > System Settings > Network > WiFi > Advanced > DNS:
Original values (before connecting):
DNS Servers: 192.168.178.1 (I cannot delete it, just add others; set via DHCP by router at home)
Search Domains: fritz.box (I cannot delete it, just add others; set via DHCP by router at home)

After connection values will change to:
DNS Servers: 192.168.1.31 (seems correct, it's the VPN server that also works as DHCP and DNS server; I cannot delete it, just add others)
Search Domains: openvpn (I don't know, probably also pushed from the VPN server; I cannot delete it, just add others)

[tunnelblick.txt](https://github.com/Tunnelblick/Tunnelblick/files/685157/tunnelblick.txt)

**Comment by jkbullard** [comment]
Everything looks OK to me. Perhaps 192.168.1.31 isn't responding.

**Comment by AniX** [comment]
192.168.1.31 is also the VPN server and DHCP server. After connecting the DHCP server is pushing the DNS server entry 192.168.1.31, so this is as expected. I also can ping 192.168.1.31 while connected.

**Comment by rippiedoos** [comment]
I also had this issue on 10.12.2 and in my case, I had multiple USB Ethernet Devices in my list and was connecting over USB Ethernet (I use multiple adapters at home an at work). 

I resolved it by removing _ALL_ USB Ethernet Devices in the list of devices in Settings -> Network and replugging my USB-device. After that it worked as before, also with my colleague. I have ticked the box 'Set DNS after routes are set' and 'Reset the primary interface after disconnecting'.

So I don't know if this is a macOS Sierra or Tunnelblick issues but I'm willing to help figure this out.

**Comment by jkbullard** [comment]
@rippiedoos - Thanks for your comment.

That sounds like a Sierra problem -- Tunnelblick has nothing to do with the Ethernet drivers.

**Comment by AniX** [comment]
I have been able to solve the issue. The same issue has occurred on a Mac that was still on 10.11.6 and has used OpenVPN all the time with no config changes or updates in the client.

I also learned that other existing OpenVPN clients (Android, Windows) had no issue. However, we use pfSense as OpenVPN server and have recently also updated to pfSense 2.3.2. It seems that the new version behaves a little different or has modified the existing configuration.

So, the problem is not a bug in Tunnelblick (or Sierra) and for this reason, I will close this issue.

In case someone with similar issues finds this, here our solution:
Under _OpenVPN Advanced Client Settings_:
Activated **Provide a default domain name to clients** and set the domain name (else it was "openvpn"). Also push DNS servers entries by activating **Provide a DNS server list to clients**, where first, we added the DNS Server in the VPN subnet, and also added to public DNS servers (e.g. 8.8.8.8, 8.8.4.4). However, this didn't help in our case. We also had to change _Client Settings > Topology_ from **net30 - isolated /30 network per client** to **Subnet - One IP address per client in a common subnet**.

The pfSense help gets more specific regarding Topology setting: _"Specifies the method used to supply a virtual adapter IP address to clients when using TUN mode on IPv4.
Some clients may require this be set to "subnet" even for IPv6, such as OpenVPN Connect (iOS/Android). Older versions of OpenVPN (before 2.0.9) or clients such as Yealink phones may require "net30"."_

So, with "net30", it worked for all clients but Mac/Tunnelblick. After switching to "subnet", it worked just fine for everyone.
